ledger/src/shred.rs

```diff
@@ -448,33 +442,68 @@ impl Shred {
         self.common_header.slot
     }
 
-    pub fn parent(&self) -> Option<Slot> {
+    pub fn parent(&self) -> Result<Slot> {
         match self.shred_type() {
             ShredType::Data => {
-                let parent_offset = Slot::try_from(self.data_header.parent_offset);
-                self.slot().checked_sub(parent_offset.ok()?)
+                // TODO What if parent_offset is zero and slot != 0?
```

Comment:
I think the call to this filter in window service will drop the shred https://github.com/solana-labs/solana/blob/0fd8e3c5d749832612f0e44c624f05a30e1ab48b/core/src/window_service.rs#L167

```
if parent_slot >= slot {
   return false;
}
```

Reply:
done